lib/xandra/cluster/control_connection.ex

```diff
@@ -412,13 +426,67 @@ defmodule Xandra.Cluster.ControlConnection do
     end
   end
 
-  defp queried_peer_to_host(%{"peer" => _} = peer_attrs) do
+  defp queried_peer_to_host(%{"rpc_address" => rpc_address} = peer_attrs, true = use_rpc_address)
+       when is_tuple(rpc_address) do
+    {address, peer_attrs} = Map.pop!(peer_attrs, "rpc_address")
+    peer_attrs = Map.delete(peer_attrs, "peer")
+    peer_attrs = Map.put(peer_attrs, "address", address)
+    queried_peer_to_host(peer_attrs, use_rpc_address)
+  end
+
+  defp queried_peer_to_host(%{"rpc_address" => _} = peer_attrs, true = use_rpc_address) do
```

Comment:
When is it that we get `rpc_address` as a string? Xandra should return `:inet.ip_address/0` for IPs, with a 4-element tuple for IPv4 and 8-element tuple for IPv6.

Reply:
When the peers table is queried, it was using the string. 

For example this was one of the logs (when address is ipv6 this was another issue because the validation for ipv6 fails in xandra)
```
[] 21:35:58.910 [info] Xandra: [:connected]: %{address: '10.4.0.12', connection_name: nil, port: 11172, protocol_module: Xandra.Protocol.V4, supported_options: %{"COMPRESSION" => ["snappy", "lz4"], "CQL_VERSION" => ["3.4.5"], "PROTOCOL_VERSIONS" => ["3/v3", "4/v4", "5/v5", "6/v6-beta"]}}
```

Now the log looks like 
```
[] 16:52:41.083 [info] Xandra Cluster: [:change_event]: %{cluster_name: {:local, FastSummariesConsumer.Store}, cluster_pid: #PID<0.5488.0>, event_type: :host_added, host: %Xandra.Cluster.Host{address: {10, 4, 0, 12},
```

There is still another place where strings are being used for the original host for the control connection if I'm not mistaken but it broke a lot of tests trying to change that out. But more than breaking tests, I think might have been more code changes needed to make it work. That would be good for a different PR to fix.

Reply:
This is the other location where string is in use https://github.com/Shimmur/xandra/blob/c563d737b99bcdec195a38408db490fec046a5e8/lib/xandra/cluster/pool.ex#L469 and I think might be good to convert this to a tuple as well.

Reply:
> When the peers table is queried, it was using the string.

I’m confused. If you look at the schema for the `system.peers` table in C* itself, `rpc_address` is of type `inet`, so how can the result of the query to that table include `rpc_address` as a string?

Reply:
I'm also adding the validate_node options. There was only one previously and it took a string https://github.com/lexhide/xandra/blob/main/lib/xandra/options_validators.ex#L51
if it were not a string, it would have failed...

Reply:
Ok, I get that, but I’m still confused on why we are handling strings from the result of a query to the `system.peers` table, which returns a `inet` as the `rpc_address`, which is in turn parsed as a tuple, not a string. Does that make sense?

Reply:
I don't know about in this particular case, but we were seeing IPv6 addresses getting treated as strings.

Reply:
@whatyouhide you are right about the query returning the inet format. 
It looks like there are some back n forth translations happening, and maybe it shouldn't be. 
https://github.com/lexhide/xandra/blob/96303f07bae820ff693d85f2e1585cb343118e00/lib/xandra/cluster/pool.ex#L464-L466 part of the code is formatting the `:node` as a string. 
e.g. 
```elixir
host = %Xandra.Cluster.Host{address: {10, 4, 0, 12}, port: 1234}

Xandra.Cluster.Host.format_address(host)
"10.4.0.12:1234"
```
so `conn_options` has a string and the `id` is a tuple. https://github.com/lexhide/xandra/blob/96303f07bae820ff693d85f2e1585cb343118e00/lib/xandra/cluster/pool.ex#L469-L471

When `validate_node` is run on the node address it is fine because it was a string.

So now I'm confused too. I do not completely follow the flow here... 

I assumed this was pulling a string and wanted to make sure it was only using a tuple. BUT then also validate node started failing (if my memory is correct) so I had to update that too. So there might be multiple (at least one more) places in the code that a connection is happening, the first one being`start_pool`...

I got the desired result but the original connection is still a string. I this it makes sense to not use `format_address` for code (maybe only for logging things). 

(I also feel like I'm talking in circles)

Reply:
@jacquelineIO okay yeah this code I have in Xandra is a little bit messy. I'll do another pass and fix this up later on then, so that we can standardize on having a single way of representing addresses.

Reply:
@whatyouhide that went in already https://github.com/lexhide/xandra/pull/328/commits/8340b4d6e20fa45d36a52f79c18893152daae983

Reply:
Ah cool awesome! Will review this soon then